Code/Editor/CryEdit.cpp

```diff
@@ -697,10 +694,12 @@ void CCryEditApp::OnFileSave()
         AZ_Assert(prefabIntegrationInterface != nullptr, "PrefabIntegrationInterface is not found.");
 
         prefabIntegrationInterface->SaveCurrentPrefab();
+
+        // when attempting to save, update the last known location using the active camera transform
+        AzToolsFramework::StoreViewBookmarkLastKnownLocationFromActiveCamera();
```

Comment:
@hultonha , SaveCurrentPrefab() saves whatever prefab you are currently 'focused' in. So if you had a level with a prefab inside it and you are focused on the prefab, it will save that prefab instead of the level. So, the last known bookmark location will be saved even in that case. 

I believe that's not the behavior you want and only want the bookmarks location to be saved on 'level save'? If that's the case, you'd need to instead put this inside PrefabIntegrationManager::SaveCurrentPrefab() and only call it if the template being saved is the root template id.
Pinging @AMZN-daimini to see if that solution makes sense or if he has any other ideas on how to solve this.

Reply:
If you are okay with camera location being saved on every ctrl+s, then what you have here works. However, that does break the paradigm that Editor team and UX established previously that you should be saving/modifying only the currently focused prefab. Saving an entity on the root prefab when your focus is on a child prefab sidesteps that paradigm and opens doors for more things to be modified that are not in the current focused prefab. This PR doesn't have to be blocked by this discussion but I strongly encourage that the viewport team seek ux guidaince on this behavior.
Pinging @rainbj for visibility on this

Reply:
> However, that does break the paradigm that Editor team and UX established previously that you should be saving/modifying only the currently focused prefab.

If I understand the system correctly, this is technically not the case. Last known location is a local bookmark, meaning that it's not saved in the prefab file but in a file that is local to the user and gitignored.
I do think we should discuss this with UX, but I think it's a new paradigm and does not necessarily need to follow the rules we use for focusing (since we won't save this info for nested prefabs)

Reply:
True, it'll be a local file but that local file will be referenced in the level container entity, making it dirty. So if the user hits ctrl+s inside a focused prefab, and exits focus, the level will now be marked dirty and the user would need to hit ctrl+s again to make the save 'permanent' or stick to the level. So I do believe that some confusion here is possible on 'what gets saved and where does the saved info reside?'

Reply:
I also checked and unfortunately @rainbj is out for this week so perhaps this is something we can sync-up on when when they're back?

Reply:
Thanks for the clarification. Looks like I got the flow wrong then. I agree that this PR doesn't have to be blocked for this discussion to happen.

Reply:
Unfortunately there was an issue with the interaction with Automated Tests I need to address - I need to do a little more investigation. It might be necessary to add a setreg option to disable 'Last Known Location' when running the tests, or better yet just have it work transparently, but it seems some tests set a camera position only to have it overwritten by the last known location value when the level finishes loading 😖 Will update when I have a fix (might be a little while when I find time again sorry)